### DISCUSSION TITLE
Can you tell me what is the fastest solution for whisper

### DISCUSSION THREAD

**Discussion by sh1man999** [k2-fsa/sherpa#703] [Q&A]
Can you tell me what is the fastest solution for whisper. Are there any models that do not have the 30-second audios limitation

https://github.com/k2-fsa/icefall/blob/master/egs/aishell/ASR/whisper/whisper_encoder_forward_monkey_patch.py#L15

**Comment by csukuangfj** [comment]
Do you want to recognize audios longer or shorter than 30s?

**Reply by sh1man999** [reply]
> Do you want to recognize audios longer or shorter than 30s?

Yes

**Reply by csukuangfj** [reply]
what does yes mean?

**Reply by sh1man999** [reply]
I want to recognize long audio files and short audio files

**Reply by sh1man999** [reply]
> what does yes mean?

I need to process long files, is there already a faytuning model ?

**Reply by csukuangfj** [reply]
please see our python-api-examples, You can use vad to process it first.

**Reply by sh1man999** [reply]
> please see our python-api-examples, You can use vad to process it first.

There are no ready-made models that solve the 30-second problem?

**Reply by csukuangfj** [reply]
You cannot find a non-streaming model anywhere accepting audio files of any length.

**Reply by sh1man999** [reply]
> You cannot find a non-streaming model anywhere accepting audio files of any length.

Why not, you say

Using Fine-tuned Whisper

Official whisper models only accept 30-second audios. To improve the throughput, you could fine-tune the whisper model to remove the 30 seconds restriction. See [examples](https://github.com/k2-fsa/icefall/blob/master/egs/aishell/ASR/whisper/whisper_encoder_forward_monkey_patch.py#L15).

We prepared two [Chinese fine-tuned whisper](https://github.com/k2-fsa/icefall/blob/master/egs/multi_zh-hans/ASR/RESULTS.md#multi-chinese-datasets-without-datatang-200h-finetuning-results-on-whisper-large-v2) TensorRT-LLM weights repo. They could be directly used from [here.](https://huggingface.co/yuekai/whisper_multi_zh_tllm_checkpoint/tree/main)

**Reply by csukuangfj** [reply]
Theoretically, if your RAM is infinite, then it is possible.

Removing the 30s constraint means you can input audio files less than 30s without padding. It does not mean you.can input files of arbitrary length.


By the way, is there any disadvantage of using a vad model here?

**Reply by sh1man999** [reply]
> Теоретически, если у вас оперативная память бесконечна, то это возможно.
> 
> Снятие ограничения на 30 секунд означает, что вы можете вводить аудиофайлы менее 30 секунд без заполнения. Это не означает, что вы можете вводить файлы произвольной длины.
> 
> Кстати, есть ли здесь какие-то недостатки использования модели vad?

Okay, I realized that long files aren't possible.
I haven't noticed any flaws in the VAD model, I didn’t use VAD enough to properly assess the audio.

How do i enter audio files less than 30 seconds without filling in ?

**Reply by csukuangfj** [reply]
Padding is invisible for users if the input audio is less than 30s. it is an implementation detail.

**Reply by sh1man999** [reply]
> Padding is invisible for users if the input audio is less than 30s. it is an implementation detail.

Faytuning the model has improved your results ?

strings x = (x + self.positional_embedding[:x.shape[1], :]).to(x.dtype)
